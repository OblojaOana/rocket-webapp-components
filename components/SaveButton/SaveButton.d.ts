export interface SaveButtonProps {

    /**
     * The text to be displayed on tooltip
     */
    title?: string;

    /**
     * Callback fired when a "click" event is detected
     * @param {object} event The event source of the callback.
     */
    onClick: (event: React.SyntheticEvent) => void

    /**
     * If set to true, the button will be disabled
     */
    disabled?: boolean

    /**
     * The color of the button
     * @default "theme"
     */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground"

    /**
     * The size of the button
     * @default "medium"
     */
    size?: "small" |
    "medium"

    /**
     * The size of the font
     * @default "small"
     */
    fontSize?: "inherit" |
    "default" |
    "small" |
    "large"

    /**
    * Demos:
    * 
    * - https://bit.dev/totalsoft_oss/react-mui/save-button
    * 
    *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/SaveButton/SaveButton.js
    */
}

export default function SaveButton(props: SaveButtonProps): JSX.Element;