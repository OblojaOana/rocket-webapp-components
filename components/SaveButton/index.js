import SaveButton from "../../core/SaveButton";
export default SaveButton;