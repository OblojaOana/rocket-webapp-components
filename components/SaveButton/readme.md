##### Copy this snapshot in the overview panel

```js
import React from "react";
import SaveButton from "@bit/totalsoft_oss.react-mui.save-button";

export default <SaveButton title="Save" />;
```
