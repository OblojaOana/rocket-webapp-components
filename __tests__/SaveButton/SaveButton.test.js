import React from "react";
import renderer from "react-test-renderer";
import SaveButton from "../../components/SaveButton/SaveButton";

describe("Render SaveButton", () => {
  it("render correctly SaveButton component", () => {
    const component = renderer.create(<SaveButton title="Save" />);

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
