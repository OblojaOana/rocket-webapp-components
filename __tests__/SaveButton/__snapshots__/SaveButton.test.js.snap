// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton Snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Save"
  className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Save"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
      />
    </svg>
  </span>
</button>
`;
